File: PianoForte/Web/App/js/enum.js
'use strict';

goog.provide('PianoForte.Enum');

PianoForte.Enum = function () {
    this['Status'] = {
        'None': 0,
        'Active': 1,
        'Inactive': 2,
        'Drop': 3,
        'Resigned': 4,
        'Available': 5,
        'Empty': 6,
        'Cancelled': 7,
        'Paid': 8,
        'NotPaid': 9,
        'Waiting': 10,
        'CheckedIn': 11,
        'Postponed': 12,
        'StudentAbsence': 13,
        'StudentMissing': 14,
        'TeacherAbsence': 15,
        'TeacherMissing': 16,
        'TeacherQuitted': 17,
        'SchoolHoliday': 18,
        'Deleted': 19
    };

    this['ContactType'] = {
        'None': 0,
        'Phone': 1,
        'Email': 2
    };
}

File: PianoForte/Enum/Enum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace PianoForte.Enum
{
    public enum ClassroomType
    {
        NONE,                   // 0
        NORMAL,                 // 1
        POSTPONE,               // 2
        EXTRA,                  // 3
    }

    public enum ProductType
    {
        NONE,                   // 0
        COURSE,                 // 1
        BOOK,                   // 2
        CD,                     // 3
        OTHER,                  // 4
    }

    public enum Status
    {
        NONE,                   // 0          
        ACTIVE,                 // 1
        INACTIVE,               // 2
        DROP,                   // 3
        RESIGNED,               // 4
        AVAILABLE,              // 5
        EMPTY,                  // 6
        CANCELLED,              // 7
        PAID,                   // 8
        NOT_PAID,               // 9
        WAITING,                // 10
        CHECKED_IN,             // 11
        POSTPONED,              // 12
        STUDENT_ABSENCE,        // 13
        STUDENT_MISSING,        // 14
        TEACHER_ABSENCE,        // 15
        TEACHER_MISSING,        // 16
        TEACHER_QUITTED,        // 17
        SCHOOL_HOLIDAY,         // 18
        DELETED                 // 19
    }

    public enum UserRole
    {
        NONE,                   // 0
        ADMIN,                  // 1
        LOCAL_ADMIN,            // 2
        LOCAL_USER,             // 3
        STUDENT,                // 4
        TEACHER                 // 5
    }

    public enum ContactType
    {
        NONE,                   // 0
        PHONE,                  // 1
        EMAIL                   // 2
    }

    public enum TeacherProperty
    {
        NONE,                   // 0
        TEACHES_45_MIN          // 1
    }   

    public class Enum
    {
        // Do Nothing
    }
}